### PR TITLE
Split app.jsx accross multiple files and upgraded to ES6.

### PR DESCRIPTION
**Pull request by nand-stripe**
Tested locally to ensure that most of the functionality is unchanged.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,40 @@
 {
   "name": "timberlake",
-  "version": "1.0.0",
+  "version": "0.0.1",
```
Comment (abull-stripe) [inline package.json]: I think this will break semantic versioning, though I doubt that anyone us using this as a library

**Comment by abull-stripe** [comment]
Other than the gulpfile and package.json is pretty much everything just moving things around?
Any specific places I should look at carefully?

We should probably switch from using gulp to webpack at some point since that's what we've standardized around mostly, but I don't think it's super worth it now.

I assume you manually QAed the site fairly carefully? (It's not that big so there isn't too much to click through)

File: package.json (hunk 1 above)
Comment (nand-stripe) [inline package.json]: Uuuups, I missed the fact that the file was already there.

Comment (nand-stripe) [inline package.json]: Ups, I missed the fact that package.json was already there. Reverted the unnecessary changes.

**Comment by nand-stripe** [comment]
I agree we should switch to webpack, but maybe later in the future since from what I see that would also require some go changes. I have upgraded from React 1.x to React 2.x, that means that the way routing is implemented changed, but functionality is the same. I have tested the functionality I could access - I could not access finished jobs, nor could I see logs of jobs.

**Comment by nand-stripe** [comment]
- gulpfile.js changed. I agree we should switch to webpack, but maybe later in the future since from what I see that would also require some go changes.
- I have upgraded from React 1.x to React 2.x, that means that the way routing is implemented changed, but functionality is the same.
- I have tested the functionality I could access - I could not access finished jobs, nor could I see logs of jobs.
